h/views/api_users.py

```diff
@@ -23,17 +25,17 @@ def create(request):
     service directly.
     """
     client = _request_client(request)
-    payload = request.json_body
 
-    user_props = {
-        'authority': client.authority,
-        'username': payload['username'],
-        'email': payload['email'],
-    }
+    schema = schemas.CreateUserAPISchema()
+    appstruct = schema.validate(request.json_body)
 
-    user_signup_service = request.find_service(name='user_signup')
-    user = user_signup_service.signup(require_activation=False, **user_props)
+    _check_authority(client, appstruct)
```

Comment:
I think this approach is absolutely fine for now, but one option that would keep the validation more self-contained would be to pass the client into the schema constructor and then override `validate`. That is, the class would look like:

```python 
class CreateUserAPISchema(JSONSchema):
    schema = ...

    def __init__(self, client):
        self.client = client
        super(CreateUserAPISchema, self).__init__()

    def validate(self, data):
        appstruct = super(CreateUserAPISchema, self).validate(data)
        if self.client.authority != data.get('authority'):
            ...
        return appstruct
```